src/backend/SyncBackendConnection.cpp

```diff
@@ -307,79 +318,60 @@ bool SyncBackendConnection::update_item_info_(
-                changed = true;
-                subentity_item = item->submodel()->find(backend::backend_id_to_models_id(subentity_id));
+                // Only create the new entity if is alive or inactive are visible
+                if (inactive_visible || get_alive(subentity_id))
+                {
+                    // Creates the Item object and update its data
+                    model->appendRow((this->*create_function)(subentity_id));
+                    changed = true;
+                    models::ListItem* subentity_item = model->find(subentity_id);
 
-                // It shold not fail after including it in row
-                assert(subentity_item);
+                    changed = update_item_(subentity_item, update_function, inactive_visible) || changed;
+                }
             }
-            changed = (this->*update_function)(subentity_item) || changed;
-        }
-    }
-    catch (const Exception& e)
-    {
-        qWarning() << "Fail updating subitems: " << e.what();
-    }
 
-    return changed;
-}
+            // In case this entity is inactive and inactive are not being displayed, remove ir
+            else if (!inactive_visible && !get_alive(subentity_id))
+            {
+                models::ListItem* subentity_item = model->at(index);
 
-bool SyncBackendConnection::update_model_(
-        ListModel* model,
-        EntityKind type,
-        EntityId id,
-        bool (SyncBackendConnection::* update_function)(ListItem*),
-        ListItem* (SyncBackendConnection::* create_function)(EntityId))
-{
-    bool changed = false;
+                // Remove the row
+                model->removeRow(index);
 
-    try
-    {
-        // For each User get all processes
-        for (auto subentity_id : StatisticsBackend::get_entities(type, id))
-        {
-            // Check if it exists already
-            models::ListItem* subentity_item = model->find(backend::backend_id_to_models_id(subentity_id));
+                // Remove its subentities and the object ListItem
+                delete subentity_item;
 
-            // If it does not exist, it creates it and add a Row with it
-            // If it exists it updates its info
-            if (nullptr == subentity_item)
-            {
-                model->appendRow((this->*create_function)(subentity_id));
                 changed = true;
-                subentity_item = model->find(backend::backend_id_to_models_id(subentity_id));
+            }
 
-                // It shold not fail after including it in row
-                assert(subentity_item);
+            // Otherwise just update the entity
+            else
+            {
+                models::ListItem* subentity_item = model->at(index);
+                changed = update_item_(subentity_item, update_function, inactive_visible) || changed;
             }
-            changed = (this->*update_function)(subentity_item) || changed;
         }
     }
     catch (const Exception& e)
```

Comment:
In general, this raises the following warning in windows: `warning C4101: 'e': unreferenced local variable`
Remove the `e` if it's not used.

Reply:
It is being used in the `catch` statement:
`qWarning() << "Fail updating model: " << e.what();`
I do not understand the warning, but I do not think it is incorrectly used.